File: src/main/java/com/github/njuro/jard/ban/UnbanForm.java
package com.github.njuro.jard.ban;

import static com.github.njuro.jard.common.InputConstraints.MAX_BAN_REASON_LENGTH;

import com.github.njuro.jard.common.Constants;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.Size;
import lombok.Builder;
import lombok.Data;

/** Form for invalidating ban before its natural expiration. */
@Data
@Builder
@SuppressWarnings("JavadocReference")
public class UnbanForm {

  /** {@link Ban#ip} */
  @NotNull(message = "{validation.ban.ip.null}")
  @Pattern(regexp = Constants.IP_PATTERN, message = "{validation.ban.ip.pattern}")
  private String ip;

  /** {@link Ban#unbanReason} */
  @Size(max = MAX_BAN_REASON_LENGTH, message = "{validation.ban.reason.max}")
  private String reason;
}

File: src/main/kotlin/com/github/njuro/jard/rewrite/ban/dto/UnbanForm.kt
package com.github.njuro.jard.rewrite.ban.dto

import com.github.njuro.jard.rewrite.ban.Ban
import com.github.njuro.jard.rewrite.common.IP_PATTERN
import com.github.njuro.jard.rewrite.common.InputConstraints.MAX_BAN_REASON_LENGTH
import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size

/** Form for invalidating ban before its natural expiration.  */
data class UnbanForm(
    /** [Ban.ip]  */
    @NotNull(message = "{validation.ban.ip.null}") @Pattern(
        regexp = IP_PATTERN,
        message = "{validation.ban.ip.pattern}"
    )
    val ip:  String?,

    /** [Ban.unbanReason]  */
    @Size(
        max = MAX_BAN_REASON_LENGTH,
        message = "{validation.ban.reason.max}"
    )
    val reason: String?
)
